### ISSUE TITLE
Bug: Tasks and columns not visible after adding tasks when running under Lighttpd

### ISSUE THREAD

**Issue by Turysaz** [kanboard/kanboard#3844]
This issue is:

- [x] Bug report
- [ ] Feature request
- [ ] Improvement

- Please, read the [Frequently Asked Questions](https://docs.kanboard.org/en/latest/developer_guide/faq.html).
    - [x] done.
- Ask questions on the [Forum](https://github.com/kanboard/forum/issues).
    - [x] not a question.

### Actual behaviour

Whithin a project board, I add a couple of tasks. After a certain amount of tasks is reached, the column is no longer displayed correctly. The last task within that column is cut off after the task number and all succeeding tasks are not displayed.
Also, when this error occurs, **no** succeeding columns are displayed.
(Screenshot 1)

When I drag the last task to the top (or middle) of the column, **sometimes** the rendering is corrected and the following tasks as well as the following columns are displayed correctly again. (Sometimes I had to do it twice or more times until the problem was gone.)
This and the fact that I could reproduce the bug easily by just adding a couple of default tasks (titles: 'a', 'b', ...) indicate that this is **not a problem with the configuration of a specific task**.

Also, if the amount of tasks grows further (10-15 tasks), it seems to reach a point where I cannot "repair" it by changing the order of the tasks. The last task is alway cut off then.

I tried to investigate the problem with the firefox inspector and found out that the succeeding columns are completely missing.

I just installed a fresh copy of Kanboard v1.2.3. The only thing I've changed to the vanilla installation is adding a new user (manager) and logging into kanboard as the new user and creating an default project (non-private). Then I started creating some tasks and after 7 tasks the problem occurred.

### Expected behaviour

All tasks are displayed correctly.

### Steps to reproduce

1. Download v1.2.3.zip
2. Extract
3. 'chown www-data:www-data ./data'
4. Open kanboard
5. (not shure if necessary) Create new user (manager), log in as new user.
6. Create project
7. Go to project board
8. Spam some tasks
9. Problem occurs after 5 to 10 tasks. (I experienced it at 5 tasks in total within a configured environment and then again after 7 tasks within a fresh copy of kanboard. Don't know if this correlates to the configuration state of the installation...)

#### Last task is cut off. Cannot drop the dragged task to 'Ready' column, because it is not existent.

![screenshot 1](https://user-images.githubusercontent.com/16538627/39473804-008acdc4-4d51-11e8-8876-5d45b33f4d97.png)

#### Hidden column is not displayed correctly after the bug happend in column 1

![screenshot 2](https://user-images.githubusercontent.com/16538627/39473805-00a4849e-4d51-11e8-8767-418938506094.png)

#### There really is no second column

![screenshot 3](https://user-images.githubusercontent.com/16538627/39473806-00bf6962-4d51-11e8-8a16-77f091d47d6d.png)

#### The first column is still there after hiding it, but the expand button still isn't.

![screenshot 4](https://user-images.githubusercontent.com/16538627/39473807-00db3a34-4d51-11e8-8c05-1b638d79785b.png)

### Logs

This is from the lighttpd error log, don't know if that has to do something about my problem:

    2018-05-01 13:51:20: (mod_fastcgi.c.2543) FastCGI-stderr: PHP Fatal error:  Uncaught PicoDb\SQLException: SQL Error: SQLSTATE[HY000]: General error: 5 database is locked in /srv/www/html/kanboard/vendor/        fguillot/picodb/lib/PicoDb/StatementHandler.php:351
    2018-05-01 13:51:20: (mod_fastcgi.c.2543) FastCGI-stderr: Stack trace:
    2018-05-01 13:51:20: (mod_fastcgi.c.2543) FastCGI-stderr: #0 /srv/www/html/kanboard/vendor/fguillot/picodb/lib/PicoDb/StatementHandler.php(245): PicoDb\StatementHandler->handleSqlError(Object(PDOException))
    2018-05-01 13:51:20: (mod_fastcgi.c.2543) FastCGI-stderr: #1 /srv/www/html/kanboard/vendor/fguillot/picodb/lib/PicoDb/Database.php(257): PicoDb\StatementHandler->execute()
    2018-05-01 13:51:20: (mod_fastcgi.c.2543) FastCGI-stderr: #2 /srv/www/html/kanboard/vendor/fguillot/picodb/lib/PicoDb/Table.php(218): PicoDb\Database->execute('UPDATE "session...', Array)
    2018-05-01 13:51:20: (mod_fastcgi.c.2543) FastCGI-stderr: #3 /srv/www/html/kanboard/app/Core/Session/SessionHandler.php(62): PicoDb\Table->update(Array)
    2018-05-01 13:51:20: (mod_fastcgi.c.2543) FastCGI-stderr: #4 [internal function]: Kanboard\Core\Session\SessionHandler->write('GKjwXHB2us-4HQK...', 'csrf|a:2:{s:60:...')
    2018-05-01 13:51:20: (mod_fastcgi.c.2543) FastCGI-stderr: #5 [internal function]: session_write_close()
    2018-05-01 13:51:20: (mod_fastcgi.c.2543) FastCGI-stderr: #6 {main}
    2018-05-01 13:51:20: (mod_fastcgi.c.2543) FastCGI-stderr:   thrown in /srv/www/html/kanboard/vendor/fguillot/picodb/lib/PicoDb/StatementHandler.php on line 351

### Configuration

- Kanboard version: v.1.2.3 ( + v1.2.2)
- Database type and version: sqlite
- Webserver: lighttpd
- PHP version: 7.0.27
- OS: raspbian 9 (raspberry pi)
- Browser:
    - Linux: Firefox 59.0.2 (64bit), qutebrowser v1.2.1
    - Windows: Firefox 59.?.? (same as Linux, I guess. Not my PC. )

**Comment by Turysaz** [comment]
Update: same problem with Kanboard v.1.2.2

**Comment by Turysaz** [comment]
**Update:** I did some more investigation and it seems as if the problems comes from the combination with Lighttpd.
I tested the setup on a Debian9 machine with Lighttpd and experienced the same problem. Then I replaced Lighttpd by Apache2 (still on the VM) and I could not reproduce the problem.
I haven't tested Apache on the Raspi, though. I'd prefer to not install Apache for I'd like to stick to a slim webserver..

**Comment by creecros** [comment]
NGINX?

**Comment by Turysaz** [comment]
Thank you for your answers!
I ended up installing apache after all and now everything works fine with that workaround.
Maybe I'll try Nginx soon, I haven't used it before.

Nevertheless I think that the locked database is not what causes the issue here or at least that my original problem is not caused by the rasperry's IO, because I experienced the same problem on my Debian-VM. The raspberry is accessing peripheral storage devices, but the VM's host has a pretty good internal hard drive.

Another observation:
Both the apache server and lighttpd are running the same instance of kanboard. (Not at the same time, of course.) When switching between the servers, the error is persistent: While I do not see any problems with apache, after switching back to lighttpd it's the same task that is broken as before switching forth and back.
